packages/elements-dev-portal/src/components/NodeContent/NodeContent.tsx

```diff
@@ -64,39 +58,37 @@ export const NodeContent = ({
   tryItCorsProxy,
 }: NodeContentProps) => {
   return (
-    <PersistenceContextProvider>
```

Comment:
@stoplightio/pierogi-platoon @mpodlasin is removing `PersistenceContextProvider` from here unsafe in any way that you can think of?

This is the root cause of https://github.com/stoplightio/elements/pull/1900#issuecomment-987265556. Basically we already wrap the entire dev portal in a `PersistenceContextProvider`, but then we're wrapping each page in it's own `PersistenceContextProvider` again here. This causes atom values accessed by children to essentially reset between pages (unless the atom is persisted somewhere other than memory, such as local storage - this is why the issue doesn't present itself for the code sample selection state).

Reply:
How did you go about figuring out that the double context provider was causing the state to reset between pages? (Just curious for future reference)

Reply:
IMO as long as we can guarantee that any other use of `NodeContent` is wrapped with `PersistenceContextProvider`(Sourcegraph would be useful to figure that out), we're good.

That being said, I _feel_ like this would count as a breaking change *iff* it's normal for non-Stoplight repos to use this component (it's `export`'d, so who knows?) without `StoplightProject`. Any consumers of `NodeContent` would have problems.

Reply:
> How did you go about figuring out that the double context provider was causing the state to reset between pages?

Since something was up w the `chosenServer` value when navigating between previously unmounted/loaded operation pages, I started by logging that value out via good old console log. I noticed that it was resetting to its default value, and from there I had a hunch that the state library must be somehow resetting or pulling from different contexts (hunch was in part a result of running into these sorts of issues in the past w state libraries, I actually don't know much about Jotai specifically - this and the recent JSV PR is first time I've used it). Last step was to look-up how Jotai manages context, and then search codebase for that jotai provider to check it it might be double rendered... anddd, whala.

> IMO as long as we can guarantee that any other use of...

A good thought - with how things seem to be organized atm, we cannot guarantee this and there is indeed a use case to render `NodeContent` (rather than `StoplightProject`, which is basically a pre-built/easy-mode combination of `NodeContent`, `TableOfContents`, etc).

However, `DevPortalProvider` always needs to be in the tree, so that could be a good spot - @mpodlasin any particular reason the persistence provider is included via the `StoplightProject` component rather than the `DevPortalProvider`? For that matter, `StoplightProject` renders a `DevPortalProvider`, and both of them include the mosaic context and react-query context (so we're doubling up on these unnecessarily). Any reason persistence, mosaic, and react-query context can't be consolidated to `DevPortalProvider`?

Reply:
I looked a bit on development history of dev-portal at it seems it just happened this way, for no particular reason.

I think moving it up the tree to `DevPortalProvider` is absolutely the most logical solution 👍